core/scale/type_encoder.hpp

```diff
@@ -17,28 +18,28 @@ namespace kagome::common::scale {
   struct TypeEncoder {
     // if required to calculate size, we can add
     // canCalculateSize() and calculateSize(const T& item) methods
-    bool encode(const T &item, Buffer &out) {
+    outcome::result<void> encode(const T &item, Buffer &out) {
```

Comment:
use T&& - 
and pass it as `std::forward<T>(item)` to other funcs

Reply:
They say it makes no sense, because there is no type deduction in this case and T&& becomes just r-value reference.